airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py

```diff
@@ -0,0 +1,210 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import urllib.parse
+from abc import ABC
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class SunshineStream(HttpStream, ABC):
+    primary_key = "id"
+    data_field = "data"
+    url_base = "https://XXX"
+    page_size = 100
+
+    def __init__(self, subdomain: str, start_date: pendulum.datetime, **kwargs):
+        self.url_base = f"https://{subdomain}.zendesk.com/api/sunshine/"  # url_base #kwargs.pop("base_url")
+        self._start_date = start_date
+        self.subdomain = subdomain
+        super().__init__(**kwargs)
+
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        delay_time = response.headers.get("Retry-After")
+        if delay_time:
+            return int(delay_time)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        resp_json = response.json()
+        if resp_json.get("links") and resp_json.get("links").get("next"):
+            next_query_string = urllib.parse.urlsplit(resp_json.get("links").get("next")).query
+            params = dict(urllib.parse.parse_qsl(next_query_string))
+            return params
+        return {}
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        return {"Content-Type": "application/json"}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        yield from response_json.get(self.data_field, [])
```

Comment:
Are there cases when response isn't dict?

Reply:
yes, mostly response is a list, but in one case it can be a dict  - this case is `ObjectTypePolicies` stream, it overrides this method and I will fix default value for it

Reply:
can you also leave a comment noting this.